.github/workflows/Nightly.yml

```diff
@@ -0,0 +1,47 @@
+# CI for Julia nightly, separate workflow to avoid failing CI badge on nightly fail
+name: Nightly
```

Comment:
oh, there's an unfortunate naming clash here.

This file is for the Nightly Version of Julia (as you say in the comment above). 
But we will also want an action that runs CI on master each night, and which notifies folks at Invenia if it fails... and unfortunately, we _also_ call that "nightly" (despite being unrelated to the Nightly Version of Julia). This is the thing to do with having cron jobs and email notification that came up before at https://github.com/JuliaDiff/ChainRules.jl/pull/309#issuecomment-730429184}

Where the unfortunateness compounds is that we probably will end up with another file for this "run the CI each night and notify us of failures" thing... which we'll probably want to call "Nightly" 🤦‍♂️ And the pain is real: here's a PR that adds a "Nightly.yml" file with this second, different meaning to the "Nightly.yml" here
https://github.com/invenia/Intervals.jl/pull/151/files

The good news is the is purely a naming clash... so let's just rename this file! 
Suggestions welcome. Clearly, naming things is difficult.

Also, sorry.

Reply:
#whatsinaname

Reply:
Maybe just tweak to JuliaNightly.yml ?

Reply:
To add to the complication, we can't have CI in the workflow name, because at some point in the past (maybe still), that would cause its status to be reflected in the CI status badge. I renamed to JuliaNightly.

Reply:
thanks! Ooof this GitHub Actions stuff is fiddly. Thank you so much for tackling it!

Reply:
No problem!